torch/onnx/symbolic_opset9.py

```diff
@@ -897,6 +894,11 @@ def batch_norm(g, input, weight, bias, running_mean, running_var, training, mome
         bias_value = torch.tensor([0.] * input_sizes[1]).type(
             'torch.' + input.type().scalarType() + 'Tensor')
         bias = g.op("Constant", value_t=bias_value)
+
+    if len(input_sizes) == 2:
+        # batchnorm1d accepts 2d and 3d array, but ONNX only accepts 3d
```

Comment:
ONNX BN seems to allow for 1D tensor, but that seems like a special case. Is that why you do not have a branch for `len(input_sizes) == 1`? Also, we should consider updating the comment here because technically ONNX BN allows 1D.

Reply:
Thanks for mentioning this. I was just thinking about the bug fix so didn't check ONNX spec. Actually looks like this case is supported. Looks like we don't need the branching here. 
It just needs a small fix in Caffe2 spatial batch norm to account for this.
Will update this.